openpype/hosts/houdini/plugins/create/create_unreal_staticmesh.py

```diff
@@ -0,0 +1,164 @@
+    label = "Unreal - Static Mesh (FBX)"
+    family = "staticMesh"
+    icon = "fa5s.cubes"
+
+    default_variants = ["Main"]
+
+    def create(self, subset_name, instance_data, pre_create_data):
+
+        instance_data.update({"node_type": "filmboxfbx"})
+
+        instance = super(CreateUnrealStaticMesh, self).create(
+            subset_name,
+            instance_data,
+            pre_create_data)
+
+        # get the created rop node
+        instance_node = hou.node(instance.get("instance_node"))
+
+        # get parms
+        parms = self.get_parms(subset_name, pre_create_data)
+
+        # set parms
+        instance_node.setParms(parms)
+
+        # Lock any parameters in this list
+        to_lock = ["family", "id"]
+        self.lock_parameters(instance_node, to_lock)
+
+    def get_network_categories(self):
+        return [
+            hou.ropNodeTypeCategory(),
+            hou.sopNodeTypeCategory()
+        ]
+
+    def get_pre_create_attr_defs(self):
+        """Add settings for users. """
+
+        attrs = super().get_pre_create_attr_defs()
+        createsubnetroot = BoolDef("createsubnetroot",
+                                   tooltip="Create an extra root for the "
+                                           "Export node when it's a "
+                                           "subnetwork. This causes the "
+                                           "exporting subnetwork node to be "
+                                           "represented in the FBX file.",
+                                   default=False,
+                                   label="Create Root for Subnet")
+        vcformat = EnumDef("vcformat",
+                           items={
+                               0: "Maya Compatible (MC)",
+                               1: "3DS MAX Compatible (PC2)"
+                           },
+                           default=0,
+                           label="Vertex Cache Format")
+        convert_units = BoolDef("convertunits",
+                                tooltip="When on, the FBX is converted"
+                                        "from the current Houdini "
+                                        "system units to the native "
+                                        "FBX unit of centimeters.",
+                                default=False,
+                                label="Convert Units")
+
+        return attrs + [createsubnetroot, vcformat, convert_units]
+
+    def get_dynamic_data(
+        self, variant, task_name, asset_doc, project_name, host_name, instance
+    ):
+        """
+        The default subset name templates for Unreal include {asset} and thus
+        we should pass that along as dynamic data.
+        """
+        dynamic_data = super(CreateUnrealStaticMesh, self).get_dynamic_data(
+            variant, task_name, asset_doc, project_name, host_name, instance
+        )
+        dynamic_data["asset"] = asset_doc["name"]
+        return dynamic_data
```

Comment:
Why is this needed? The default publish templates also have `{asset}` in it and work just fine.
I'm pretty sure this is redundant. I see `asset` in the folder and file templates for `render`, `publish`, `hero`, etc.
What's so special here?

Reply:
tbh, I was imitating unreal static mesh creator in maya
subset name became dynamic only after:
1. adding the `"asset"` to the dynamic data 
2. adding subset name template 
![image](https://github.com/ynput/OpenPype/assets/20871534/bb5a5073-a140-45fd-99dc-227f90d84070)

Reply:
Still - usage of `{asset}` and `{variant}` wouldn't require the need to override `get_dynamic_data`. Right @iLLiCiTiT ?

Reply:
I gave it a shot and it doesn't work without passing `{asset}` as dynamic data.

Reply:
Oh man - I clearly wasn't awake. `{asset}` is a default key for folder and file templates but **not** for the subset name formatting. That's why this needs to be passed along in `get_dynamic_data` explicitly.